### DISCUSSION TITLE
Image component has two wrapper divs that can't be styled or removed

### DISCUSSION THREAD

**Discussion by AleksandrHovhannisyan** [vercel/next.js#18820] [Help]
# Bug report

## Describe the bug

The `next/image` component renders unnecessary wrapper `<div>`s, with no API to style these or remove them from the DOM. This leads to:

1. Accessibility errors/HTML validation errors. Divs cannot be children of certain elements (e.g., buttons). Images can be.
2. Layout issues. It's not possible to position the images where you want them because of the divs.

## To Reproduce

1. Create a simple Next.js app with no dependencies.
2. Import and render the `next/image` component.
3. Inspect the DOM and note that there are wrapper divs around the `<img`> element.

## Expected behavior

Ideally, the wrapper divs should not be there. A `<picture>` element with nested `<img>` and `<source>`s would make more sense, with an exposed API to allow us to customize the picture parent and the nested image. If this is not possible, there should at least be props for the wrapper divs so we can position them accordingly.

## Screenshots

![image](https://user-images.githubusercontent.com/19352442/98179716-62398d80-1ecd-11eb-9af6-73cf72a814bc.png)

## System information

- OS: Windows
- Browser (if applies): Chrome
- Version of Next.js: 10.0.0

## Additional context

See discussion [here, on the Image component RFC](https://github.com/vercel/next.js/discussions/16832#discussioncomment-118288)

**Comment by Timer** [comment]
Have you tried Next.js 10.0.1? There's only one wrapping div now.

**Reply by AleksandrHovhannisyan** [reply]
Nope, I haven't but I'll give that a shot. In any case, that still poses the same issue of an unstylable wrapper parent.

**Comment by THPubs** [comment]
Can't we have an option where there's no wrapping divs. Just an img tag but still does the image optimisation (webp and image sizes) and lazy loading? Currently the Image tag can only be used for specific use cases only. Yesterday I upgraded to 10.0.1 and it's layout option is not an alternative to unsized. When I changed to layout, all the images broke (Became larger than the container size).

**Reply by luknl** [reply]
Is it the same issue we encountered [here](https://github.com/vercel/next.js/discussions/18739#discussioncomment-121272)?

**Reply by THPubs** [reply]
Yeap. That's the same issue I faced. 👍

**Reply by luknl** [reply]
Any news about that @THPubs? Have you found a way to manage this?

**Comment by estebanmunchjones2019** [comment]
Hi all! good news! The internal wrapping divs CAN be styled. Here's an example how:
Here is a solution that works for accessing the first internal wrapping div of the <Image> element

```
// JSS styling here //
imageWrapper: {
    "& div:first-child": {
      position: "static !important"
    }
  }
```

```
//JSX code here//   
<div className={classes.imageWrapper}>
   <Image src={image1} layout="responsive" width={1450} height={750} alt="First slide" />
</div>
```

**Reply by AleksandrHovhannisyan** [reply]
While this does work, I don't prefer this solution since it introduces another unnecessary wrapper div around two existing divs. Imo, the simplest solution is for Next.js to expose props for the parent div so we can at least style it (but the best solution may be to get rid of the div altogether—Gatsby doesn't use any wrapper divs, for example). In certain situations, divs are invalid children anyway, so this limits the usage of Next.js's image component to just block contexts, for example.

**Reply by estebanmunchjones2019** [reply]
yes, makes sense, would be nice to have props to do it

**Comment by farskid** [comment]
is there a way to at least use a different element for the wrapper element? Using MDX and `next/image` for all images in the markdown, I get invalid HTML since the wrapper div is a descendant of paragraph elements.

**Comment by danielbonifacio** [comment]
We need a prop like `disableWrapperElements` for solving this. I don't think that an Image optimizer component should inferrer in styling.

**Reply by im-ironclad** [reply]
+1 on this, i'm really disliking this component lately because it hijacks and uglifies my markup. Not to mention it makes styling a headache.

**Reply by DevJett** [reply]
Its really does not make sense to have wrapper `span`s or `div`s if you don't want to use the `layout` prop

**Reply by tonioriol** [reply]
Came here to say that. Please don't wrap the img, it makes styling a pain.

**Comment by nachtfunke** [comment]
Are there any updates on this? This behaviour makes this component essentially unusable.

**Comment by AleksandrHovhannisyan** [comment]
I think this discussion can be closed. As of Next.js v13, the Image component no longer renders a wrapper, so the img tag can be styled directly. See also: https://github.com/vercel/next.js/discussions/22861
